### PR TITLE
fix addTransactions bug

### PR DESCRIPTION
**Pull request by mask-pp**

### CODE REVIEW

File: core/types/minorblock.go [hunk 1]
```diff
@@ -57,7 +58,10 @@ func (m *MinorBlockMeta) Hash() common.Hash {
 // Hash returns the block hash of the header, which is simply the keccak256 hash of its
 // Serialize encoding.
 func (h *MinorBlockHeader) Hash() common.Hash {
-	return serHash(*h, nil)
+	if h.hash == (common.Hash{}) {
```
Comment (pingke) [inline core/types/minorblock.go]: need to check, the head content may change

Comment (pingke) [reply core/types/minorblock.go]: head may copy

File: cluster/slave/slave_grpc.go [hunk 2]
```diff
@@ -696,7 +696,7 @@ func (s *SlaveServerSideOp) AddTransactions(ctx context.Context, req *rpc.Reques
 
 	if gReq.Branch != 0 {
 		err := addTxList(gReq.Branch, txs.TransactionList)
-		return new(rpc.Response), err
```
[the remark this answers is not in the document]
Comment (mask-pp) [reply cluster/slave/slave_grpc.go]: changed.

File: cluster/slave/api_backend.go [hunk 3]
```diff
@@ -425,38 +425,38 @@ func (s *SlaveBackend) getMinorBlockHeadersWithSkip(gReq *p2p.GetMinorBlockHeade
...
+	headerlist := make([]*types.MinorBlockHeader, 0, gReq.Limit)
 	for len(headerlist) < cap(headerlist) && height >= 0 && height <= mTip.NumberU64() {
 		iHeader := shrd.MinorBlockChain.GetHeaderByNumber(height)
 		if qcom.IsNil(iHeader) {
 			break
 		}
-		headerlist = append(headerlist, iHeader.(*types.MinorBlockHeader))
 		if gReq.Direction == qcom.DirectionToGenesis {
 			height -= uint64(gReq.Skip) + 1
+			headerlist = append([]*types.MinorBlockHeader{iHeader.(*types.MinorBlockHeader)}, headerlist...)
```
[the remark this answers is not in the document]
Comment (mask-pp) [reply cluster/slave/api_backend.go]: changed.

**Pull request merged**